### PR TITLE
Project Survey - Week 6 Julia, Inna, Susanne

### PR DESCRIPTION
**Pull request by JuliaHolm**
Pair project @JuliaHolm, @InnaKokic and @smExlex (Susanne Ekenheim).

https://willowy-meerkat-5ea81b.netlify.app/

### CODE REVIEW

File: src/components/ChooseCat/ChooseCat.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  return (
+    <form className="choose-cat-form">
+      <h3>Choose your cat</h3>
+      {/* This div with classname flex is only for my eyes sake (Julia) */}
+      <div className="cat-image-container">
+        {/* Mapping over the 'catOptions' array. The 'Key' prop is used to uniquely identify each cat. */}
+        {catOptions.map((catOption) => (
+          <div key={catOption.id} className="each-cat-wrapper">
+            {/* Using label element to improve accssibility and user experience. The 'Key' prop is used to uniquely identify each label element and it's is important for React's virtual DOM to efficiently track and update elements. */}
```
Comment (AnnaRobertsson) [inline src/components/ChooseCat/ChooseCat.jsx:17]: Well done!

File: src/components/NextQuestion/NextQuestion.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    username: "",
+    gender: "",
+    age: "",
+    personality: "",
+    chosenCat: "",
+    catName: "",
+  });
+  // This function updates the form data and continue to the next question.
+  const handleNextQuestion = (data) => {
+    //  This is a functional update. It takes the previous state 'prevData', spreads it's properties '...prevData', and then spreads the properties of the new data. This effectively merges the previous form data with the new data.
```
Comment (AnnaRobertsson) [inline src/components/NextQuestion/NextQuestion.jsx:17]: Informative!

File: src/components/ChooseCat/ChooseCat.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            <label>
+              <input
+                type="radio"
+                className="radio-input"
+                value={catOption.id}
+                checked={chooseCat === catOption.id}
+                onChange={handleChooseCats}
+              />
+              <div className="radio-custom-img">
+                <img src={catOption.imgSrc} alt={catOption.label} />
```
Comment (AnnaRobertsson) [inline src/components/ChooseCat/ChooseCat.jsx:27]: Really fun to use the pictures as a radio button! Good job!

File: src/components/StartPage/StartPage.jsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import "./StartPage.css";
+
+export const StartPage = ({ handleNextQuestion }) => {
+  return (
+    <div className="start-wrapper">
+      <div className="right-column">
+        <h1>Adopt a Cat</h1>
+        <h2>Find your fur-ever pet!</h2>
+        {/* When the button is clicked it will trigger the handleNextQuestion function */}
+        <button onClick={handleNextQuestion} className="adopt-btn">
```
Comment (AnnaRobertsson) [inline src/components/StartPage/StartPage.jsx:10]: Really nice to have a welcoming start page and then display the questions.

**Review by AnnaRobertsson — APPROVED** [review]
First of all - WOW! The design is really nice and in every step of the survey it looks good. I have also looked at it in different screen sizes and it really looks great and neat in all sizes. Very good job!

Everything seems to be working well and I could use the keyboard to navigate through the whole survey. It was a bit hard to tell that you were actually on the buttons for the cat's gender or age for example, since they didn't get the black border as the rest of the buttons did. But by pressing the arrows instead of tab you could choose between the options. At text size 200%, all the text is easily read and everything still looks in place, it’s just the background-image that becomes very enlarged.

The code review was difficult with the +6600 changed files, and of course I understand that this wasn't something that you strived for. It was really hard to find the relevant files to look through and comment. Thank you for so quickly looking into it when I informed you it looked a bit weird in the pull request, and fixing it!

Be really proud of your awesome project!

File: src/App.jsx [hunk 5]
```diff
@@ -1,3 +1,177 @@
...
+                    <img src="/prev.png" alt="" />
+                  </button>
+                )}
+                {currentQuestion < 6 && currentQuestion !== 0 && (
+                  <button className="next-btn" onClick={handleNextQuestion}>
+                    <img src="./next.png" alt="" />
+                  </button>
+                )}
+
+                {/* 'aria-label' is an accessability attribute that provides an accessible label for screen readers, informing them about the purpose of the button. */}
```
[the remark this answers is not in the document]
Comment (verawitting) [reply src/App.jsx:127]: +1!

File: src/components/Age/Age.jsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+// Rendering a form with radio buttons to select the age of the cat.
+export const Age = ({ age, setAge }) => {
+  const handleAge = (event) => {
+    setAge(event.target.value);
+  };
+  return (
+    <form
+      className="age-form"
+      onSubmit={(event) => event.preventDefault()}
+      required
```
Comment (verawitting) [inline src/components/Age/Age.jsx:12]: Great to have a required attribute! Although it doesn't seem to work? :o

**Review by verawitting — COMMENTED** [review]
I agree with Anna on most comments! Great code!! Fantastic CSS, as always Julia! I liked to see the useState used as much as you did, made me understand it even better! I think you have a good structure, clear variables! I think you got the accessibility pretty good! There was one thing do, I couldn't tab myself through to the second option of the radio button, maybe I'm just doing it wrong. But great great job!

**Pull request closed without merging**